src/handlers/setDependencyVersionHandler.ts

```diff
@@ -0,0 +1,198 @@
+// Licensed under the MIT license.
+
+import * as fse from "fs-extra";
+import * as vscode from "vscode";
+import { mavenExplorerProvider } from "../explorer/mavenExplorerProvider";
+import { Dependency } from "../explorer/model/Dependency";
+import { MavenProject } from "../explorer/model/MavenProject";
+import { constructDependenciesNode, constructDependencyNode, getIndentation } from "../utils/editUtils";
+import { UserError } from "../utils/errorUtils";
+import { ElementNode, getNodesByTag, XmlTagName } from "../utils/lexerUtils";
+import { getVersions } from "../utils/requestUtils";
+
+const CONFLICT_INDICATOR: string = "omitted for conflict";
+
+export async function setDependencyVersionHandler(selectedItem?: Dependency): Promise<void> {
+    if (selectedItem === undefined) {
+        throw new UserError("No dependency node specified.");
+    }
+    const pomPath: string = selectedItem.projectPomPath;
+    if (!await fse.pathExists(pomPath)) {
+        throw new UserError("Specified POM file does not exist on file system.");
+    }
+
+    let effectiveVersion: string;
+    if (selectedItem.supplement.indexOf(CONFLICT_INDICATOR) !== -1) {
+        const re = /\(omitted for conflict with ([\w.-]+)\)/gm;
+        effectiveVersion = selectedItem.supplement.replace(re, "$1");
+    } else {
+        effectiveVersion = selectedItem.version;
+    }
+
+    const gid: string = selectedItem.groupId;
+    const aid: string = selectedItem.artifactId;
+    const versions: string[] = getAllVersionsInTree(pomPath, gid, aid);
+    const searchCommand: string = "Search versions from Maven Central Repository...";
+    versions.push(searchCommand);
+
+    const selectedVersion: string | undefined = await vscode.window.showQuickPick(
+        versions.map(version => ({ value: version, label: version !== searchCommand ? `$(package) ${version}` : version, description: version === effectiveVersion ? "effective" : ""})),
+        { placeHolder: `Select a version for ${gid}:${aid}...`,
+            ignoreFocusOut: true}
+    ).then(version => version ? version.value : undefined);
+    if (!selectedVersion) {
+        return;
+    }
+    if (selectedVersion === searchCommand) {
+        const selectedVersionFromOrigin: string | undefined = await vscode.window.showQuickPick<vscode.QuickPickItem & { value: string }>(
+            getVersions(gid, aid).then(artifacts => artifacts.map(artifact => ({value: artifact.v, label: `$(package) ${artifact.v}`, description: artifact.v === effectiveVersion ? "effective" : "" }))),
+            { placeHolder: `Select a version for ${gid}:${aid}...`,
+                ignoreFocusOut: true }
+        ).then(artifact => artifact ? artifact.value : undefined);
+        if (!selectedVersionFromOrigin) {
+            return;
+        }
+        if (selectedVersionFromOrigin !== effectiveVersion) {
+            await setDependencyVersion(pomPath, gid, aid, selectedVersionFromOrigin);
+        }
+    } else {
```

Comment:
nested if-else loop with multipe place applying changes. The logic can be simplified as below.
```
let selectedVersion;  // sv for short
sv = first quickpick
if (sv === undefined) return;
if (sv === OPTION_MAVEN_CENTRAL)   sv = second quick pick from maven central items
if (sv !== effectiveVersion)  applyChanges


```